### PR TITLE
Add DAgger and DART imitation learning algorithms

### PR DESCRIPTION
**Pull request by PedroLucas63**
# Summary
This pull request extends the imitation learning stack in CORL with two new JAX-based algorithms, DAgger and DART, while improving core training infrastructure used by BC and expert-driven workflows. It also updates environment handling for parallel rollout collection, adds reproducible test entrypoints for PPO and imitation pipelines, and introduces `uv` as the project package/environment manager.

# Motivation
The existing BC workflow trains only from a static offline dataset. For locomotion and sim-to-real style scenarios, this limits adaptation because the policy is not exposed to states induced by its own mistakes.

These changes address that gap by:
- adding online dataset aggregation (DAgger) to reduce compounding error;
- adding disturbance-aware aggregation (DART) to improve robustness under action perturbations;
- enabling configurable parallel actors in environment creation to collect data efficiently;
- improving developer and CI ergonomics via `uv`-based dependency resolution and lockfile support.

# Changes Introduced
## Core updates
- Fixed warning-prone typing/default patterns in `algorithms/offline/bc_jax.py`.
- Updated `get_env` in `algorithms/utils/wrapper_gym.py` to accept `num_actors` and pass it through to the wrapper.
- Updated BC training/evaluation flow in `algorithms/offline/bc_jax.py` with improved type annotations, safer normalization/evaluation behavior, optional model loading, and config serialization using `asdict`.

## New algorithms
- Added `algorithms/offline/dagger_jax.py`.
- Added `algorithms/offline/dart_jax.py`.

Both implementations follow the existing BC-style supervised actor update pattern and reuse shared conventions (dataset normalization, Flax train state, JIT update loops, checkpointing, and optional video logging).

## Testing additions
- Added `test/ppo_jax.py` for expert policy checkpoint loading and evaluation.
- Added `test/offline/bc_jax.py` for evaluating BC-family policies (BC/DAgger/DART checkpoints) including optional rendering and video export.

## Dependency and project configuration
- Reworked `pyproject.toml` into a `project` + `tool.uv` layout.
- Declared runtime dependencies directly in `pyproject.toml`.
- Added a pinned `uv.lock` for reproducible environments.
- Added explicit PyTorch CUDA index and git source for `mujoco_playground` (`playground`).

## Repository hygiene
- Updated `.gitignore` to ignore generated training artifacts (`videos/`, `logs/`).
- Existing ignore entries for checkpoint-related directories remain in place (`checkpoints`, `expert_checkpoints`, etc.).

# Implemented Algorithms
## DAgger (Dataset Aggregation)
`algorithms/offline/dagger_jax.py` implements an iterative imitation loop:
1. Train a policy on an initial offline dataset.
2. Roll out with a behavior policy that mixes expert and learner actions.
3. Query the expert on visited states to relabel actions.
4. Aggregate new labeled transitions into the dataset.
5. Retrain and repeat for `dagger_iterations`.

Integration details:
- Uses PPO expert checkpoint loading through Brax/MuJoCo Playground interfaces.
- Supports parallel collection via `collect_num_envs` and the new `get_env(..., num_actors=...)` API.
- Uses an exponentially decayed expert mixing probability (`decay_factor`) to transition from expert-guided behavior to learner-driven behavior.

## DART (Disturbances for Augmented Robot Training)
`algorithms/offline/dart_jax.py` implements disturbance-based data aggregation:
1. Train an initial policy from offline data.
2. Estimate action-noise covariance from expert-policy mismatch on sampled dataset points.
3. Collect trajectories using expert actions plus sampled perturbations.
4. Label with expert actions, aggregate transitions, and retrain iteratively.

Integration details:
- Reuses the BC-style actor architecture and training loop.
- Adds `estimate_noise(...)` and per-iteration perturbation sampling (`dart_samples_for_noise_estimation`).
- Uses the same environment/evaluation/checkpoint/video infrastructure as BC and DAgger.

# Environment Setup
Install dependencies and create the environment with `uv`:

```bash
uv sync
```

This uses `pyproject.toml` + `uv.lock` to install pinned dependencies, including CUDA-enabled JAX/PyTorch and project-specific sources.

# Running Experiments
Example commands using `uv`:

```bash
# BC
uv run -m algorithms.offline.bc_jax

# DAgger
uv run -m algorithms.offline.dagger_jax

# DART
uv run -m algorithms.offline.dart_jax
```

# Tests
Run test/evaluation scripts with `uv`:

```bash
# PPO expert checkpoint evaluation
uv run -m test.ppo_jax

# BC-family policy evaluation (BC / DAgger / DART checkpoints)
uv run -m test.offline.bc_jax
```

You can adapt CLI flags (environment name, episode count, rendering, command type) based on the scenario.

# Notes
- Training artifacts such as generated rollout videos and logs are now ignored by git (`videos/`, `logs/`).
- Checkpoint-related directories were already covered by ignore rules and remain excluded.
- The branch introduces substantial new algorithm code and a lockfile (`uv.lock`), so reviewers should focus on algorithmic flow, data aggregation logic, and checkpoint compatibility across BC/DAgger/DART.

### CODE REVIEW

**Comment by PedroLucas63** [comment]
Olá, @luanagbmartins!
Abri esse PR com a implementação do DAgger e o DART. Poderia realizar a análise quando tiver algum tempo?
